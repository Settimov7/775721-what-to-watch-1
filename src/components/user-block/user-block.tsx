import * as React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';

import {getIsAuthorizationRequired, getUserAvatarSrc} from '../../reducer/user/selectors';

interface Props {
  isAuthorizationRequired: boolean;
  userAvatarSrc: string | null;
}

export const UserBlock: React.FunctionComponent<Props> = (props) => {
  const {isAuthorizationRequired, userAvatarSrc} = props;

  return(
    <div className="user-block">
      {isAuthorizationRequired ? (
        <Link to="/login" className="logo__link">Sign in</Link>
      ) : (
        <Link to="/mylist">
          <div className="user-block__avatar">
            <img src={userAvatarSrc} alt="User avatar" width="63" height="63"/>
          </div>
        </Link>
      )}
    </div>
  );
};

const mapStateToProps = (state) => ({
  isAuthorizationRequired: getIsAuthorizationRequired(state),
  userAvatarSrc: getUserAvatarSrc(state),
});

export default connect(mapStateToProps)(UserBlock);
