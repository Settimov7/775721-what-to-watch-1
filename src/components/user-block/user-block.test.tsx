import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';

import {UserBlock} from './user-block';

it(`User block correctly renders if authorization required`, () => {
  const props = {
    isAuthorizationRequired: true,
    userAvatarSrc: null,
  };
  const tree = renderer.create(
    <BrowserRouter>
      <UserBlock {...props}/>
    </BrowserRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`User block correctly renders if authorization not required`, () => {
  const props = {
    isAuthorizationRequired: false,
    userAvatarSrc: `image.jpg`,
  };
  const tree = renderer.create(
    <BrowserRouter>
      <UserBlock {...props}/>
    </BrowserRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
