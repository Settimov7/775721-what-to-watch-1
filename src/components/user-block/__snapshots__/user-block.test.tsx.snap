// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User block correctly renders if authorization not required 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <img
      alt="User avatar"
      height="63"
      src="image.jpg"
      width="63"
    />
  </div>
</div>
`;

exports[`User block correctly renders if authorization required 1`] = `
<div
  className="user-block"
>
  <a
    className="logo__link"
    href="/login"
    onClick={[Function]}
  >
    Sign in
  </a>
</div>
`;
